lib/database.js

```diff
@@ -89,5 +89,50 @@ function getCoursesFromUserId(id, callback) {
   });
 }
 
+/**
+ *
+ * @param {number} courseId - course id
+ * @param {function} callback - Called on success or error returns (err, result)
+ * result object consists of many rows, has structure:
+ * [
+ *  {
+ *    course_name: string,
+ *    course_description: string,
+ *    start_dtm: string,
+ *    end_dtm: string,
+ *    prof_fname: string,
+ *    prof_lname: string,
+ *    prof_email: string
+ *  },
+ *  ...
+ * ]
+ */
+function getCourseMetaData(courseId, callback) {
+  pool.getConnection((err, connection) => {
+    if (err) {
+      logger.error(err.message);
+      callback(err);
+    } else {
+      connection.query("SELECT c.name as course_name, " +
+        "c.description as course_description, c.start_dtm, c.end_dtm, " +
+        "prof.fname as prof_fname, prof.lname as prof_lname, " +
+        "prof.email as prof_email " +
+        "FROM courses c INNER JOIN " +
+          "( SELECT u.fname, u.lname, u.email, lcu.course_id " +
+          "FROM lkp_course_users lcu " +
+          "INNER JOIN users u ON u.id = lcu.user_id " +
+          "WHERE u.user_type_id = 3 AND lcu.course_id = ? " +
+        ") prof " +
+        "ON prof.course_id = c.id WHERE c.id = ?",
+        [courseId, courseId],
+        (err, result) => {
+          handleResult(err, callback, result);
+          connection.release();
+        });
+    }
+  });
+}
+
+module.exports.getCourseMetaData = getCourseMetaData;
```

Comment:
This is typically how it is done unless you do something like this:

```
exports.getIdAndHashFromEmail = function (args...) { ... }
```

In this fashion you are both exporting and defining the function at the same time using an anonymous function.